SparkPost/2web2ui
# SA-1614: Compare By Form

### What Changed
 - Added compare by form 

### How To Test
 - Enable compare by feature `allow_compare_by`
 - Click `Add Comparators`
 - Select filter type
- Verify form works as expected
 - Verify data structure on submission (console log)

### To Do
- [ ] Fix inconsistent behavior for typeaheads
